addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js

```diff
@@ -19,7 +19,21 @@ export class ProductsWidgetControlPanel extends Component {
         this.notification = useService("pos_notification");
         this.orm = useService("orm");
         this.updateSearch = debounce(this.updateSearch, 100);
-        this.state = useState({ mobileSearchBarIsShown: false });
+        this.state = useState({ mobileSearchBarIsShown: false, isMobile: false });
+        this.productsWidgetControl = useRef("products-widget-control");
+        this.ui = useService("ui");
+
+        onMounted(() => {
+            this.toggleIsMobile();
+            window.addEventListener("resize", this.toggleIsMobile.bind(this));
+        });
```

Comment:
```suggestion
        const toggle = () => this.toggleIsMobile();
        onMounted(() => {
            this.toggleIsMobile();
            window.addEventListener("resize", toggle);
        });
        onWillUnmount(() => {
            window.removeEventListener("resize", toggle);
        });
```

I don't see the issue in using only `ui.isSmall`. But if it's really problematic, the use of `onMounted` is not enough. You need to make sure the resize event listener is removed during unmount.